src/components/TimelineList/TimelineList.js

```diff
@@ -0,0 +1,412 @@
+    onTemporalMarkerHover({ hovering, marker })
+  }
+
+  /**
+   * Sets up useGesture handlers for the temporal range
+   */
+  const bindTemporalRangeGestures = useGesture({
+    onHover: handleTemporalRangeHover
+  })
+
+  /**
+   * Sets up useGesture handlers for the temporal start selection handle
+   */
+  const bindTemporalStartGestures = useGesture({
+    onHover: (state) => handleTemporalMarkerHover('start', state)
+  })
+
+  /**
+   * Sets up useGesture handlers for the temporal end selection handle
+   */
+  const bindTemporalEndGestures = useGesture({
+    onHover: (state) => handleTemporalMarkerHover('end', state)
+  })
+
+  if (!timelineWrapperRef.current) return null
+
+  const focusedRangeStyle = {}
+  const focusedRangeMaskLeftStyle = {}
+  const focusedRangeMaskRightStyle = {}
+  const temporalStartStyle = {}
+  const temporalEndStyle = {}
+  const temporalRangeStyle = {}
+  const temporalRangeMouseOverStyle = {}
+
+  // If a position is set for the temporal selection mouseover indicator, set the left property to that value
+  if (temporalRangeMouseOverPosition) {
+    temporalRangeMouseOverStyle.left = temporalRangeMouseOverPosition
+  }
+
+  const { end, start } = temporalRange
+  const { end: focusedIntervalEnd, start: focusedIntervalStart } = focusedInterval
+
+  // If a time is set for the start and end, set the left property to that temporal position
+  const timelineWrapperWidth = timelineWrapperRef.current.getBoundingClientRect().width
+
+  if (start) {
+    temporalStartStyle.left = (
+      getPositionByTimestamp({
+        timestamp: start,
+        timeIntervals,
+        zoomLevel,
+        wrapperWidth: timelineWrapperWidth
+      })
+    )
+  }
+
+  if (end) {
+    temporalEndStyle.left = (
+      getPositionByTimestamp({
+        timestamp: end,
+        timeIntervals,
+        zoomLevel,
+        wrapperWidth: timelineWrapperWidth
+      })
+    )
+  }
+
+  if (end || start) {
+    const range = [
+      start || timeIntervals[0],
+      end || generateEndTime(timeIntervals, zoomLevel)
+    ]
+
+    const [startPosition, endPosition] = range
+
+    // Style the temporal range area
+    temporalRangeStyle.left = getPositionByTimestamp({
+      timestamp: startPosition,
+      timeIntervals,
+      zoomLevel,
+      wrapperWidth: timelineWrapperWidth
+    })
+    temporalRangeStyle.width = determineScaledWidth(
+      endPosition - startPosition,
+      zoomLevel,
+      timelineWrapperWidth
+    )
+  }
+
+  if (focusedIntervalStart && focusedIntervalEnd) {
+    // Style the focused area
+    focusedRangeStyle.left = getPositionByTimestamp({
+      timestamp: focusedIntervalStart,
+      timeIntervals,
+      zoomLevel,
+      wrapperWidth: timelineWrapperWidth
+    })
+    focusedRangeStyle.width = determineScaledWidth(
+      focusedIntervalEnd - focusedIntervalStart,
+      zoomLevel,
+      timelineWrapperWidth
+    ) + 2 // Accomodate for the 2px border on the interval
```

Comment:
I assume this is in stylesheet and that would make it impossible for this to be pulled out into a variable? (I seem to remember discussing this previously)

Reply:
I think that is correct